include/libdnf/conf/option_string_list.hpp

```diff
@@ -83,19 +91,26 @@ class OptionStringContainer : public Option {
 
     /// Parses input string and returns result.
     /// @replaces libdnf:conf/OptionStringList.hpp:method:OptionStringList.fromString(const std::string & value)
-    ValueType from_string(const std::string & value) const;
+    ValueType from_string(std::string value) const;
 
     /// Converts input value to the string.
     /// @replaces libdnf:conf/OptionStringList.hpp:method:OptionStringList.toString(const ValueType & value)
     std::string to_string(const ValueType & value) const;
 
+    /// Returns the default delimiters
+    static const char * get_default_delimiters() noexcept;
+
+    /// Return delimiters of this OptionStringList
+    const char * get_delimiters() const noexcept;
+
 protected:
     /// Tests new container item value and throws exception if the item value is not allowed.
     void test_item(const std::string & item) const;
 
     std::optional<std::regex> regex_matcher;
     std::string regex;
     bool icase;
+    std::unique_ptr<const char *> delimiters;
```

Comment:
Are you using `unique_ptr` to store the raw pointer? The problem remains here. Only the pointer is copied, not the string itself. The raw pointer stored in `unique_ptr` points to the original string.

The solution is to use `std::string` instead of `const char*` to store: `std::unique_ptr<std::string> delimiters;`. 

Since it is a private part, we can even simplify it by using `std::optional<std::string> delimiters;` (`OptionBool` comes from libdnf4, it is in C++11 and `std::optional` is from C++17. Therefore `OptionBool` uses `std::unique_ptr`).

Other lines of code will need to be changed as well.
`return delimiters ? *delimiters : get_default_delimiters();` ->
`return delimiters ? delimiters->c_str() : get_default_delimiters();`

And we can also pass `delimiters` as `std::string` in the constructor. Example of solution:
```
template <typename T>
OptionStringContainer<T>::OptionStringContainer(
    const ValueType & default_value, std::string regex, bool icase, std::string delimiters)
    : Option(Priority::DEFAULT),
      regex(std::move(regex)),
      icase(icase),
      delimiters(std::move(delimiters)),
      default_value(default_value),
      value(this->default_value) {
    test(this->default_value);
}

template <typename T>
OptionStringContainer<T>::OptionStringContainer(const OptionStringContainer & src)
    : Option(src),
      regex(src.regex),
      icase(src.icase),
      delimiters(src.delimiters),
      default_value(src.default_value),
      value(src.value) {}

template <typename T>
OptionStringContainer<T> & OptionStringContainer<T>::operator=(const OptionStringContainer & src) {
    assert_not_locked();

    if (this == &src) {
        return *this;
    }
    regex = src.regex;
    icase = src.icase;
    delimiters = src.delimiters;
    default_value = src.default_value;
    value = src.value;
    return *this;
}
```

Reply:
ooh, I wasn't thinking, my bad.
Updated again hopefully without any major bugs this time.
Thank you.